cli/registry/client/endpoint.go

```diff
@@ -30,32 +31,39 @@ func (r repositoryEndpoint) BaseURL() string {
 }
 
 func newDefaultRepositoryEndpoint(ref reference.Named, insecure bool) (repositoryEndpoint, error) {
+	repoName := reference.TrimNamed(ref)
 	repoInfo, _ := registry.ParseRepositoryInfo(ref)
-	endpoint, err := getDefaultEndpointFromRepoInfo(repoInfo)
+	indexInfo := repoInfo.Index
+
+	endpoint, err := getDefaultEndpoint(ref, !indexInfo.Secure)
```

Comment:
I recall now that I was looking whether this one should just get `insecure` (the argument) passed, which is set using the `--insecure` flag on `docker manifest inspect`, but , while related, they are technically different;


- `indexInfo.Secure` is based on whether the registry is on a loopback address
- `--insecure` allows overriding a secure registry to be accessed insecurely (which, if the registry is using TLS allows for an invalid TLS connection).

It's really a bit of a kludge altogether 💩 🔥  (it was added as a workaround because the CLI doesn't know about the daemon configuration, which has the "insecure-registries" option).